cms/release_calendar/forms.py

```diff
@@ -39,18 +40,32 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
             self.fields["release_date"].disabled = True
             self.fields["status"].disabled = True
 
+        if self.instance.live_status in LOCKED_STATUS_STATUSES:
+            # If the status is locked, disable the status field
+            self.fields["status"].disabled = True
+
```

Comment:
thinking of Adam's comment, what about something like:

```python
if (
    self.instance.live_notice
    and not user_can_modify_notice(self.for_user)
):
    self.fields["notice"].disabled = True
```

?
(an perhaps make `live_notice` a cached property)

Other than this thought, LGTM